arlas-core/src/main/java/io/arlas/server/utils/ParamsParser.java

```diff
@@ -105,6 +111,21 @@ public static Aggregation getAggregationModel(List<String> agg) throws ArlasExce
         return aggregationModel;
     }
 
+    private static HitsFetcher getHitsFetcher(String fetchHitsString) throws ArlasException {
+        HitsFetcher hitsFetcher = new HitsFetcher();
+        if (StringUtil.isNullOrEmpty(fetchHitsString)) {
+            throw new BadRequestException("fetchHits should not be null nor empty");
+        }
+        Pattern p = Pattern.compile("(\\d*)(\\()(.*)(\\))");
```

Comment:
If the pattern is static, you shouldn't compile it each time but it should be a static property

Reply:
Absolutely